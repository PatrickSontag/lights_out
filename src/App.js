import React from "react";
import Board from "./Board";
import "./App.css";

/** Simple app that just shows the LightsOut game. */

function App() {
  return (
    <div className="App">
      <h1>LIGHTS OUT</h1>
      <Board nrows={3} ncols={4} chanceLightStartsOn={25} />
    </div>
  );
}

export default App;
