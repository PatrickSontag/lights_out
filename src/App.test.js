import { render, screen } from '@testing-library/react';
import App from './App';

it("should render without crashing", function() {
  render(<App/>);
})

it('should render Lights Out title', () => {
  render(<App />);
  const lightsOutTitle = screen.getByText(/lights out/i);
  expect(lightsOutTitle).toBeInTheDocument();
});

it("should match snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
})