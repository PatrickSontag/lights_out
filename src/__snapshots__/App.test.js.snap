// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      LIGHTS OUT
    </h1>
    <table
      id="game-board"
    >
      <tbody>
        <tr>
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
        </tr>
        <tr>
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell Cell-lit"
          />
        </tr>
        <tr>
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell Cell-lit"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
